submissions/Ant-C-tech/memory-pair-game/js/app.js

```diff
@@ -0,0 +1,278 @@
+    congratulationIco.setAttribute('alt', 'mentor')
+    congratulationIco.classList.add('congratulationIco')
+
+    const congratulation = document.createElement('h3')
+    congratulation.textContent = 'Congratulation!!!'
+    congratulation.classList.add('congratulation')
+
+    const congratulationText = document.createElement('p')
+    congratulationText.innerHTML = `It took you <span class="attempts">${attempts}</span> attempts!`
+    congratulationText.classList.add('congratulationText')
+
+    const congratulationSubText = document.createElement('p')
+    congratulationSubText.textContent = `Now, you maybe feel yourself true ninja or maybe even samurai... But remember...`
+    congratulationSubText.classList.add('congratulationSubText')
+
+    const newGameBtn = document.createElement('h3')
+    newGameBtn.textContent = 'A samurai has no goal, only a path...'
+    newGameBtn.classList.add('newGame')
+
+    congratulationWrapper.append(congratulationIco, congratulation, congratulationText, congratulationSubText, newGameBtn)
+
+    newGameBtn.addEventListener('click', function() {
+        TASK = []
+        attempts = 0
+        setTask()
+        window.addEventListener('resize', phoneAdaptation)
+        phoneAdaptation()
+        changeContent(createGameField())
+    }, { once: true })
+
+    return congratulationWrapper
+}
+
+function changeContent(content, callback) {
+    MAIN.classList.add('main-hide')
+    MAIN.addEventListener('transitionend', function() {
+        MAIN.innerHTML = ''
+        MAIN.append(content)
+        if (callback) {
+            callback()
+        }
+        MAIN.classList.remove('main-hide')
+    }, { once: true })
+}
+
+function showElem(elem, effect, delay) {
+    if (delay) {
+        elem.classList.add(ANIMATE_DELAY)
+    } else {
+        elem.classList.remove(ANIMATE_DELAY)
+    }
+    elem.style.display = 'block'
+    elem.classList.add(effect)
+    elem.addEventListener('animationend', function() {
+        elem.classList.remove(effect)
+    }, { once: true })
+}
+
+function hideElem(elem, effect, delay) {
+    if (delay) {
+        elem.classList.add(ANIMATE_DELAY)
+    } else {
+        elem.classList.remove(ANIMATE_DELAY)
+    }
+    elem.classList.add(effect)
+    elem.addEventListener('animationend', function() {
+        elem.style.display = 'none'
+        elem.classList.remove(effect)
+    }, { once: true })
+}
+
+function checkAnswer() {
+    MAIN.classList.add('main-notActive')
+    const openCards = document.querySelectorAll('.card-rotate')
+    if (openCards.length === 2) {
+        if (openCards[0].getAttribute('data-value') === openCards[1].getAttribute('data-value')) {
+            for (const card of openCards) {
+                card.classList.add('played')
+                const timeOut = setTimeout(() => {
+                    card.classList.add(CARDS_HIDE_ANIMATION)
+                    card.addEventListener('animationend', function() {
```

Comment:
Maybe it would be better to add this event listeners once at the beginning of creating the layout for this cards.